lib/engine/ability/tile_lay.rb

```diff
@@ -5,13 +5,14 @@
 module Engine
   module Ability
     class TileLay < Base
-      attr_reader :hexes, :tiles, :free, :discount
+      attr_reader :hexes, :tiles, :free, :discount, :special_lay
 
-      def setup(hexes:, tiles:, free: false, discount: nil)
+      def setup(hexes:, tiles:, free: false, discount: nil, special_lay: true)
```

Comment:
i'd prefer special_lay: nil

and then @special_lay ||= true

Reply:
ah yes i see in your code you check both nil and special lay, this is error prone